tutorials/scripting/c_sharp/diagnostics/GD0203.rst

```diff
@@ -0,0 +1,56 @@
+GD0203: The delegate signature of the signal must return void
+=============================================================
+
+====================================  ======================================
+                                      Value
+====================================  ======================================
+**Rule ID**                           GD0203
+**Category**                          Usage
+**Fix is breaking or non-breaking**   Breaking - If the return type is changed
+
+                                      Non-breaking - If the ``[Signal]`` attribute is removed
+**Enabled by default**                Yes
+====================================  ======================================
+
+Cause
+-----
+
+A delegate annotated with the ``[Signal]`` attribute has a return type when
+``void`` was expected.
+
+Rule description
+----------------
+
+Every signal must return ``void``. There can be multiple callbacks registered
```

Comment:
(Not really a change suggestion) This is a also a thing with normal C# events, tho I am not sure how this is resolved there.

Reply:
I believe when invoking C# events with a return value it just takes the value from the last callback, but I don't think it's very useful so probably nobody uses it.